gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/rest/api/service/v4/impl/validation/EndpointGroupsValidationServiceImpl.java

```diff
@@ -59,6 +61,9 @@ public List<EndpointGroup> validateAndSanitize(List<EndpointGroup> endpointGroup
                 validateEndpointGroupType(endpointGroup.getType());
                 validateServices(endpointGroup.getServices());
                 validateEndpointsExistence(endpointGroup);
+
+                endpointService.validateConnectorConfiguration(endpointGroup.getType(), endpointGroup.getSharedConfiguration());
```

Comment:
Having a closer look, I think something is still not good. The `validateConnectorConfiguration` method allows for a `null` value. It means that, if the endpoint configuration inheritance is disabled, it's still possible to have a `null` configuration right? Does it mean the gateway will throw an exception in that case? Why not adapt the http-proxy endpoint to protect itself against `null` configuration whatever it comes from the group or endpoint?

Reply:
True,  `endpointService.validateConnectorConfiguration` ignore null value. Does it neam the config is required at both level EndpointGroup and endpoint whatever is the inherit value ?

I can add mechanism in the http-proxy (all proxy :thinking: ?) to have defensive behaviour but returning an error at API level is still valid to inform the API publisher that the definition is not valid IMHO.